src/libs/kata-types/src/config/drop_in.rs

```diff
@@ -213,3 +213,205 @@ mod toml_tree_ops {
         }
     }
 }
+
+mod drop_in_directory_handling {
+    use crate::config::TomlConfig;
+    use std::fs;
+    use std::io::{self, Result};
+    use std::path::{Path, PathBuf};
+
+    fn get_dropin_dir_path(base_cfg_file_path: &Path) -> Result<PathBuf> {
+        let mut dropin_dir = base_cfg_file_path.to_path_buf();
+        if !dropin_dir.pop() {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "base cfg file path too short",
+            ));
+        }
+        dropin_dir.push("config.d");
+        Ok(dropin_dir)
+    }
+
+    fn update_from_dropin(base_config: &mut toml::Value, dropin_file: &fs::DirEntry) -> Result<()> {
+        if !dropin_file.file_type()?.is_file() && !dropin_file.file_type()?.is_symlink() {
```

Comment:
Double check if you need `is_symlink`; in many languages a symlink to a regular file will pass both "is file" and "is symlink" checks.

Reply:
Yes, you're right and this is an actual bug - good catch, thanks for that! A symlink to a directory would be handled inappropriately by this code.